src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperOutput.cs

```diff
@@ -24,57 +24,64 @@ internal TagHelperOutput(string tagName)
             Attributes = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
         }
 
-        // Internal for testing
-        internal TagHelperOutput(string tagName, [NotNull] IDictionary<string, string> attributes)
-            : this(tagName, attributes, string.Empty)
-        {
-        }
-
         /// <summary>
         /// Instantiates a new instance of <see cref="TagHelperOutput"/>.
         /// </summary>
         /// <param name="tagName">The HTML element's tag name.</param>
         /// <param name="attributes">The HTML attributes.</param>
-        /// <param name="content">The HTML element's content.</param>
-        public TagHelperOutput(string tagName,
-                               [NotNull] IDictionary<string, string> attributes,
-                               string content)
+        public TagHelperOutput(string tagName, [NotNull] IDictionary<string, string> attributes)
         {
             TagName = tagName;
-            Content = content;
             Attributes = new Dictionary<string, string>(attributes, StringComparer.OrdinalIgnoreCase);
+            PreContent = string.Empty;
+            _content = string.Empty;
+            PostContent = string.Empty;
         }
 
         /// <summary>
         /// The HTML element's tag name.
         /// </summary>
         /// <remarks>
-        /// A whitespace value results in no start or end tag being rendered.
+        /// A whitespace or <c>null</c> value results in no start or end tag being rendered.
         /// </remarks>
-        public string TagName
+        public string TagName { get; set; }
+
+        /// <summary>
+        /// The HTML element's pre content.
+        /// </summary>
+        /// <remarks>Value is prepended to the <see cref="ITagHelper"/>'s final output.</remarks>
+        public string PreContent { get; set; }
+
+        /// <summary>
+        /// The HTML element's content.
```

Comment:
update this comment to reflect `PreContent` and `PostContent` since `Content` doesn't provide the entire element body (when `TagName` is a real name).  might want to call this the "element's main content" ?